module/zfs/arc.c

```diff
@@ -6900,6 +6900,16 @@ arc_state_multilist_index_func(multilist_t *ml, void *obj)
 	    multilist_get_num_sublists(ml));
 }
 
+#define	WARN_TUN_FMT "ignoring tunable %s (using %llu instead)"
+#define	WARN_IF_TUN_IGNORED(tun, val) do { \
+	if ((tun) && ((tun) != (val))) { \
+		if (verbose) \
+			cmn_err(CE_WARN, WARN_TUN_FMT, (#tun), (val)); \
+		else \
+			dprintf("WARNING: "WARN_TUN_FMT"\n", (#tun), (val)); \
+	} \
+} while (0)
```

Comment:
I don't think there's a need to keep the `dprintf` case.  This would only be specifically enabled by developers with a debug build when investigating an issue, in which case they're already probably adding their own debug instrumentation. How about simplifying this to:

```c
#define WARN_IF_TUNING_IGNORED(tuning, value, do_warn) do {     \
        if ((do_warn) && (tuning) && ((tuning) != (value))) {   \
                cmn_err(CE_WARN,                                \
                    "ignoring tunable %s (using %llu instead)", \
                    (#tuning), (value));                        \
        }                                                       \
} while (0)
```
```

Reply:
That makes sense.  To be clear, the proposal here is to only retain the user visible log message when the value is ignored.   Which is inline with your suggestion, so I think we're all on the same page.